### PR TITLE
build: add option to build v8 with GN

### PR DESCRIPTION
**Pull request by hashseed**
This PR depends on https://chromium-review.googlesource.com/c/v8/v8/+/860007

### CODE REVIEW

File: configure [hunk 1]
```diff
@@ -55,6 +55,10 @@ sys.path.insert(0, 'tools')
 import getmoduleversion
 from gyp_node import run_gyp
 
+# imports in deps/v8/toos/node
```
Comment (bnoordhuis) [inline configure]: Typo: tools - somewhat superfluous comment, though.

File: configure [hunk 2]
```diff
@@ -1037,6 +1048,12 @@ def configure_v8(o):
     o['variables']['test_isolation_mode'] = 'noop'  # Needed by d8.gyp.
   if options.without_bundled_v8 and options.enable_d8:
     raise Exception('--enable-d8 is incompatible with --without-bundled-v8.')
+  if options.without_bundled_v8 and options.build_v8_with_gn:
+    raise Exception('--build-v8-with-gn is incompatible with --without-bundled-v8.')
```
Comment (bnoordhuis) [inline configure]: Can you keep this <= 80 columns?

File: configure (hunk 1 above)
Comment (hashseed) [reply configure]: Done. Added comment just for consistency :)

File: configure (hunk 2 above)
Comment (hashseed) [reply configure]: Done. Though there are quite a few >80 column lines in this file.

**Pull request closed without merging**